### PR TITLE
Add button for exporting a theme from the demo site

### LINKED ISSUE #144: Add button for exporting a theme from the demo site
The demo site could benefit from an option that allows you to export a theme to PHP code with the click of a button.

The button would only need to show up if the user has changed the theme colors in the advanced options.

The output may look something like this:

```php
[
    "background" => "#020200",
    "border" => "#a8a8a8",
    "stroke" => "#a8a8a8",
    "ring" => "#e41d44",
    "fire" => "#e41d44",
    "currStreakNum" => "#f9dd3c",
    "currStreakLabel" => "#f9dd3c",
    "sideNums" => "#5cadc0",
    "sideLabels" => "#5cadc0",
    "dates" => "#ed7b25",
]
```

There should be a button to copy the array to the clipboard.

The idea is to make it easier to contribute to [themes.php](https://github.com/DenverCoder1/github-readme-streak-stats/blob/main/src/themes.php)

### PR DESCRIPTION
**Pull request by komen205**
## Description

Hey! I created this working solution, please let me know if it's appropriated. This is not a **final solution**.

I created json_decode.php file so JS sends the properties json and php exports it to the text area.
It was the only working solution that I could make it work.
HTML and CSS is not final, please let me know where you want me to export it.

Let me know what changes I can make and if I'm going the right path.

Fixes #144

### Type of change

- [ ] Bug fix (added a non-breaking change which fixes an issue)
- [X] New feature (added a non-breaking change which adds functionality)
- [ ] Updated documentation (updated the readme, templates, or other repo files)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- [X] Tested locally with a valid username
- [X] Tested locally with an invalid username
- [ ] Ran tests with `composer test`
- [ ] Added or updated test cases to test new features

## Checklist:

- [X] I have checked to make sure no other [pull requests](https://github.com/DenverCoder1/github-readme-streak-stats/pulls?q=is%3Apr+sort%3Aupdated-desc+) are open for this issue
- [X] The code is properly formatted and is consistent with the existing code style
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X ] My changes generate no new warnings

## Screenshots

image

### CODE REVIEW

**Comment by DenverCoder1** [comment]
I'm having trouble testing this since it seems the code is no longer on the comparing branch.

Maybe try reopening this with a new branch?

image

File: src/demo/json_decode.php [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?php declare (strict_types = 1);
+$_POST = json_decode(file_get_contents("php://input"), true);
+var_export($_POST['json'],false);
+?>
```
Comment (DenverCoder1) [inline src/demo/json_decode.php:4]: There should be no need for fetching any webpages to output the export data.

All it needs to do is create a string in the required format, which can be done with JavaScript code.

Comment (komen205) [reply src/demo/json_decode.php:4]: I was thinking about that, but don't we have to format the string to PHP format? Like change the : to =>

Comment (DenverCoder1) [reply src/demo/json_decode.php:4]: Yeah, but there's no reason you can't use JavaScript to make the string have `=>` in it.

Comment (DenverCoder1) [reply src/demo/json_decode.php:4]: Consider the following code:

```js
const props = {"property":"#123123", "property2":"#123456"}
const output = "[\n" + Object.keys(props).map((key) => `\t"${key}" => "${props[key]}",\n`).join("") + "]";
console.log(output);
```

You may want to change it up a bit, but basically in a way like this, you can convert JSON into a string that resembles PHP array format.

**Pull request closed without merging**